frontend/templates/frontend/launch.html

```diff
@@ -334,6 +339,7 @@
 				field.style.display = "none";	
 				if(project_presets[choice.value] != -1) {
 					load_preset(project_presets[choice.value]);
+					preselect_last_project()
```

Comment:
This call should be earlier in the flow of execution, probably right when the page is loaded around line 1343. This way, the project presets are loaded for the preselected project. With the current code, the presets are loaded for the default selected project, and then the selected project is changed to the last project used. Otherwise, it works great, nice job.

Reply:
Yes, you are right! The project is taking a bit of time to get selected from the localStorage, due to the reason mentioned by you. But, when I called the function when the page is getting loaded (around line 1343), the same issue persists. I figured out that it is because the presets are not loaded by JavaScript but by HTML template instead. I am talking about the below set of lines. 

https://github.com/div-yam/CalcUS/blob/a3a58226b1d14020e8d4163f5a0e6ea2539f952a/frontend/templates/frontend/launch.html#L1020-L1025

I tried to find a way to use localStorage variables in the HTML template, but apparently, we can't (if what I found on the internet is right). Since JavaScript is a client-side language, and Django is server-side. Basically, we have to do something like
```
{% if proj.name == localStorage.getItem("lastSelectedProject") %}
<option name="{{ proj }}" selected>{{ proj }}</option>
```
Do, you have any recommendations on how to implement this? I have scanned every StackOverflow article regarding this issue, but still no help :(

Reply:
Yes, getting the localStore in the template when initially rendering the page will be very difficult. Instead, we can load the last selected project from localStorage (if it is there), then refresh the selected project and project preset. The trouble could be that there is a small delay between changing the project in javascript and being able to fetch the new value from the page in Javascript.

The function `project_selection_changed` is only called by the project `select`, so it would be easy to modify. You could make it so that that function takes as argument the name of the new project. The `onchange` event of the project `select` could (probably?) pass the new value as argument, which would keep it working. And now the `preselect_last_project` function could call `project_selection_changed` directly with the proper project name. This would be a pretty clean solution if everything works out.

Reply:
On working out, I found out that `onchange` would not be able to pass the new value as an argument because even after removing the `project_selection_changed` function, projects were getting rendered. So, the only option (in my opinion) is to add the options dynamically using JavaScript. But, this approach seems naive to me. I have pushed the changes. You can review them, and let me know if they are fine or if we need to find some other way.

Reply:
I don't quite understand why having projects rendered when loading the page is a problem. It is possible to change the selected option after all the options have been loaded. For example, you could use jQuery to get the last selected project:
`opt = $("#calc_project > option[name='" + localStorage.getItem("lastSelectedProject") +"']");`

The function `project_selection_changed` could take in `choice` as parameter (instead of getting it from the page). The project select would then have the property: `onchange="project_selection_changed(this.options[this.selectedIndex]);"`, which would directly pass the option object.

You fix would work, but it would be even better if the projects were rendered in the template directly instead of being added dynamically. Let's give the above another try.

Reply:
Yes, I understood the point! Sorry for being so stubborn. I have pushed the changes, and now the project presets are loaded with the preselected projects. Thanks for the help!

Reply:
Great job, everything works as desired!